widget/calendar.go

```diff
@@ -0,0 +1,247 @@
+}
+
+func (c *calendar) hideOverlay() {
+	overlayList := c.canvas.Overlays().List()
+	overlayList[0].Hide()
+}
+
+func (c *calendar) monthYear() string {
+	return time.Month(c.month).String() + " " + strconv.Itoa(c.year)
+}
+
+func (c *calendar) calendarObjects() []fyne.CanvasObject {
+	textSize := float32(8)
+	columnHeadings := []fyne.CanvasObject{}
+	for i := 0; i < 7; i++ {
+		j := i + 1
+		if j == 7 {
+			j = 0
+		}
+
+		var canvasObject fyne.CanvasObject = canvas.NewText(strings.ToUpper(time.Weekday(j).String()[:3]), color.NRGBA{0xFF, 0xFF, 0xFF, 0xBF})
+		canvasObject.(*canvas.Text).TextSize = textSize
+		canvasObject.(*canvas.Text).Alignment = fyne.TextAlignCenter
+		columnHeadings = append(columnHeadings, canvasObject)
+	}
+	columnHeadings = append(columnHeadings, c.daysOfMonth()...)
+
+	return columnHeadings
+}
+
+// CreateRenderer returns a new WidgetRenderer for this widget.
+// This should not be called by regular code, it is used internally to render a widget.
+func (c *calendar) CreateRenderer() fyne.WidgetRenderer {
+	c.monthPrevious = widget.NewButtonWithIcon("", theme.NavigateBackIcon(), func() {
+		c.month--
+		if c.month < 1 {
+			c.month = 12
+			c.year--
+		}
+		c.monthLabel.ParseMarkdown(c.monthYear())
+
+		c.dates.Objects = c.calendarObjects()
+	})
+	c.monthNext = widget.NewButtonWithIcon("", theme.NavigateNextIcon(), func() {
+		c.month++
+		if c.month > 12 {
+			c.month = 1
+			c.year++
+		}
+		c.monthLabel.ParseMarkdown(c.monthYear())
+
+		c.dates.Objects = c.calendarObjects()
+	})
+
+	c.monthLabel = widget.NewRichTextFromMarkdown(c.monthYear())
+
+	nav := container.New(layout.NewBorderLayout(nil, nil, c.monthPrevious, c.monthNext),
+		c.monthPrevious, c.monthNext, container.NewCenter(c.monthLabel))
+
+	c.dates = container.New(newCalendarLayout(32), c.calendarObjects()...)
+
+	dateContainer := container.NewVBox(nav, c.dates)
+
+	return widget.NewSimpleRenderer(dateContainer)
+}
+
+// NewCalendar creates a calendar instance
+func NewCalendar(cT time.Time, callback func(time.Time)) *calendar {
+	c := &calendar{day: cT.Day(), month: int(cT.Month()), year: cT.Year(), calendarTime: cT, callback: callback}
+	c.ExtendBaseWidget(c)
+
+	return c
+}
+
+// NewCalendarAtPos creates a calendar instance and shows at given position
+func NewCalendarAtPos(canvas fyne.Canvas, position fyne.Position, calendarTime time.Time, callback func(time.Time)) *calendar {
+	c := &calendar{
+		canvas:       canvas,
+		calendarTime: calendarTime,
+
+		day:   calendarTime.Day(),
+		month: int(calendarTime.Month()),
+		year:  calendarTime.Year(),
+
+		callback: callback}
+
+	c.ExtendBaseWidget(c)
+	widget.ShowPopUpAtPosition(c, canvas, position)
+
+	return c
+}
+
+// ShowAtPos creates a pop up with instantiated calendar
+func (c *calendar) ShowAtPos(canvas fyne.Canvas, pos fyne.Position) {
```

Comment:
This needs to be removed as it isn’t how layouts in Fyne work. It is the parent container and layout that controls where components are located. You should not say to the widgets where it should be drawn.